Doc update for userns in 1.28 (#41908)
kubernetes/website: content/en/docs/tasks/configure-pod-container/user-namespaces.md

<!-- overview -->
{{< feature-state for_k8s_version="v1.25" state="alpha" >}}

> This page shows how to configure a user namespace for pods. This allows to

Reviewer: ```suggestion
This page shows how to configure a user namespace for pods. This allows you to
```

Author: Applied

* The node OS needs to be Linux
* You need to exec commands in the host
* You need to be able to exec into pods

> * Feature gate `UserNamespacesSupport` need to be enabled

Reviewer: ```suggestion
* You need to enable the `UserNamespacesSupport`
  [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
```

Author: Fixed

> Note that the feature gate was previously named
> `UserNamespacesStatelessPodsSupport` when only stateless pods were supported.
> This feature gate is not recognized anymore.

Reviewer: ```suggestion
{{< note >}}
The feature gate to enable user namespaces was previously named
`UserNamespacesStatelessPodsSupport`, when only stateless pods were supported.
Only Kubernetes v1.25 through to v1.27 recognise `UserNamespacesStatelessPodsSupport`; for
Kubernetes `{{< skew currentVersion >}}`, you use `UserNamespacesSupport`.
{{< note >}}
```

Author: Fixed

* The node OS needs to be Linux
* You need to exec commands in the host
* You need to be able to exec into pods
* You need to enable the `UserNamespacesSupport`
  [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)

{{< note >}}
The feature gate to enable user namespaces was previously named
`UserNamespacesStatelessPodsSupport`, when only stateless pods were supported.
Only Kubernetes v1.25 through to v1.27 recognise `UserNamespacesStatelessPodsSupport`; for

> Kubernetes `{{< skew currentVersion >}}`, you need to use `UserNamespacesSupport`.

Author: I have removed everything after the `;`. This way, this sentence is always correct.
